dags/iop/parser.py

```diff
@@ -0,0 +1,258 @@
+            if reffered_id is None:
+                continue
+            institutions = article.findall(
+                f"front/article-meta/contrib-group/aff[@id='{reffered_id.get('rid')}']/institution"
+            )
+            countries = article.findall(
+                f"front/article-meta/contrib-group/aff[@id='{reffered_id.get('rid')}']/country"
+            )
+            for institution, country in zip(institutions, countries):
+                affiliations.append(
+                    {
+                        "value": ",".join([institution.text, country.text]),
+                        "country": country.text,
+                    }
+                )
+            author = {
+                "surname": surname,
+                "given_names": given_names,
+                "affiliations": affiliations,
+            }
+            authors.append(author)
+        return authors
+
+    # TODO: do I need it?
+
+    def _get_collab():
+        pass
+
+    def _get_date(self, date):
+        return datetime.date(
+            day=int(date.find("day").text),
+            month=int(date.find("month").text),
+            year=int(date.find("year").text),
+        ).isoformat()
+
+    def _get_date_published(self, article):
+        date = (
+            article.find("front/article-meta/history/date[@date-type='published']")
+            or article.find("front/article-meta/history/date[@date-type='epub']")
+            or article.find("front/article-meta/history/pub-date[@pub-type='ppub']")
+            or article.find("front/article-meta/pub-date")
+        )
+        if date:
+            return self._get_date(date)
+        else:
+            # In the worst case we return today
+            return datetime.date.today().isoformat()
+
+    def _get_journal_year(self, article):
+        date = self._get_date_published(article)
+        dt = datetime.datetime.strptime(date, "%Y-%m-%d")
+        return dt.year
+
+    def _construct_license(self, url, license_type, version):
+        if url and license_type and version:
+            return {
+                "url": url,
+                "license": f"CC-{license_type}-{version}",
+            }
+
+    def _get_license(self, article: ET.Element):
+        licenses = []
+        license_texts = article.findall(
+            "front/article-meta/permissions/license/license-p/ext-link"
+        )
+        urls = article.findall("front/article-meta/permissions/license")
+        license_urls = []
+        for license_url in urls:
+            license_urls.append(
+                license_url.attrib["{http://www.w3.org/1999/xlink}href"]
+            )
+        for license_url, license_text in zip(license_urls, license_texts):
+            if license_url:
+                url_parts = license_url.split("/")
+                clean_url_parts = list(filter(bool, url_parts))
+                version = clean_url_parts.pop()
+                licencse_type = clean_url_parts.pop()
```

Comment:
A typo `licencse_type` -> `license_type`

Suggestion if it is sure that last two elements are type and version respectively.
```suggestion
                version = clean_url_parts[-1]
                licencse_type = clean_url_parts[-2]
```